Expensify/App
# Fix up Validation Step

cc @aldo-expensify @kevinksullivan 

Still in a draft I will write up test steps tomorrow (and also test my changes).

### Details

There are a few changes introduced here, but mostly cleaning up this step so that:

- Each input has it's own validation (this wasn't a blocker, but easy enough to fix up)
- Backend errors are not shown inline anymore and instead inside a modal (I know this sounds backwards to what we will [_eventually_ want](https://github.com/Expensify/Expensify/issues/176726), but this step is to make them more consistent with how all the other back end errors are working at the moment i.e. they are shown in a modal and that's it. The validate step is the only step where they are also shown inline - so I removed that inconsistency).
- One interesting case is that the submit button is disabled if we have a `maxAttemptsReached` to handle when the page loads. I'm not sure if we _still_ want to prevent the submit button in that case, but leaving it for now as it's probably something @pecanoro will run into for [this issue](https://github.com/Expensify/App/issues/5182). 

### Fixed Issues (Related to)
https://github.com/Expensify/App/issues/4785

### Tests // QA Steps
1. Create a new account on a non public domain
2. Create a workspace and navigate to it
3. Press "Get Started" button
4. Move through the entire VBA flow to create a test account in the PENDING state
<details>
<summary>Additional Instructions for Step 4</summary>

To create a PENDING bank account (this state means we're waiting for the user to enter the 3 validation amounts he received):

Add Bank Account (note: in order to access this flow on NewDot, use the URL staging.new.expensify.com/bank-account):

Click Connect Manually

Routing Number: 011401533

Account number: 1111222233331111

Company Info (Enter anything unless specified):

Legal Business Name: Alberta Bobbeth Charleson

Company website: Enter any valid formatted website

Tax ID number: 123456789 (any nine digits)

Industry classification code: 123456 (any six digits)

Correct Expensify password

Confirm company is not on list of restricted businesses

Save & Continue

Requestor Information (Enter anything unless specified):

First name: Alberta

Second name: Charleson

Toggle confirmation you are authorized to use company bank account

Beneficial Owners:

Toggle terms and conditions

Toggle confirmation that the information provided is true and accurate
</details>

5. On the validation step enter no amounts
6. Verify a modal appears and each field is highlighted with outline and message
![2021-09-16_07-30-01](https://user-images.githubusercontent.com/32969087/133658120-9a5dc362-7d55-49e8-8694-39eb78bec25f.png)
![2021-09-16_07-30-37](https://user-images.githubusercontent.com/32969087/133658145-ed55b864-705e-4618-8779-8fd3984c26e5.png)
7. Enter invalid amounts that are numbers e.g. `123` for each field and submit
8. Verify that a modal appears instructing the user of the incorrect amounts
![2021-09-16_07-30-57](https://user-images.githubusercontent.com/32969087/133658207-f8d8d15a-7c01-47c0-84b3-be71e8c785a8.png)
9. Keep repeating the bad amounts and pressing submit until you get blocked.
10. A message should appear on the screen and you should **not** be able to continue once locked out.
![2021-09-16_07-33-01](https://user-images.githubusercontent.com/32969087/133658446-9b4eaa42-68f9-4930-83dd-c56b1eac8185.png)
11. Press the contact us link and verify you are brought to the 1:1 DM with Concierge.

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS
![2021-09-16_08-32-49](https://user-images.githubusercontent.com/32969087/133666284-9a04b060-4ba7-4c9f-a460-3d47fdeb2aa9.png)
![2021-09-16_08-33-01](https://user-images.githubusercontent.com/32969087/133666305-24d59012-71ec-4421-a1f4-144c14fef03f.png)
![2021-09-16_08-33-32](https://user-images.githubusercontent.com/32969087/133666316-1d20fd0d-1f43-41a2-b107-1cc3902e422a.png)

#### Android
![2021-09-16_09-19-27](https://user-images.githubusercontent.com/32969087/133672477-0e8b6c07-0f02-4e9c-a789-19b2a9655b25.png)
![2021-09-16_09-18-31](https://user-images.githubusercontent.com/32969087/133672479-f9add9b1-2f30-4b5b-8f9f-fe273faf1e9d.png)
![2021-09-16_09-18-16](https://user-images.githubusercontent.com/32969087/133672480-c7d74d21-81dd-4afa-a525-9af5b18badaa.png)
![2021-09-16_09-18-11](https://user-images.githubusercontent.com/32969087/133672482-f070102a-37f6-4d61-a2b6-7fd70c3d8185.png)
